pkg/volume/csi/csi_attacher.go

```diff
@@ -285,12 +277,49 @@ func (c *csiAttacher) MountDevice(spec *volume.Spec, devicePath string, deviceMo
 	if spec == nil {
 		return fmt.Errorf("attacher.MountDevice failed, spec is nil")
 	}
-	csiSource, err := getCSISourceFromSpec(spec)
+
+	var (
+		volumeHandle     string
+		driverName       string
+		fsType           string
+		nodeStageSecrets map[string]string
+		volAttribs       map[string]string
+	)
+
+	volSrc, pvSrc, err := getSourceFromSpec(spec)
 	if err != nil {
 		klog.Error(log("attacher.MountDevice failed to get CSI persistent source: %v", err))
 		return err
 	}
 
+	if utilfeature.DefaultFeatureGate.Enabled(features.CSIInlineVolume) && volSrc != nil {
+		if volSrc.VolumeHandle == nil {
```

Comment:
If you change `assertVolumeHandle` to only allow persistent volume types, then I think you can replace this check.